src/switchroot/ostree-mount-util.h

```diff
@@ -50,58 +54,33 @@ static inline char *
 read_proc_cmdline (void)
 {
   FILE *f = fopen("/proc/cmdline", "r");
-  char *cmdline = NULL;
-  size_t len;
-
   if (!f)
-    goto out;
-
-  /* Note that /proc/cmdline will not end in a newline, so getline
-   * will fail unelss we provide a length.
-   */
-  if (getline (&cmdline, &len, f) < 0)
-    goto out;
-  /* ... but the length will be the size of the malloc buffer, not
-   * strlen().  Fix that.
-   */
-  len = strlen (cmdline);
+    return NULL;
 
-  if (cmdline[len-1] == '\n')
-    cmdline[len-1] = '\0';
-out:
-  if (f)
-    fclose (f);
-  return cmdline;
+  char buf[COMMAND_LINE_SIZE] = {'\0'};
+  size_t len = fread(&buf, 1, sizeof (buf), f);
+  fclose(f);
+  return strndup(buf, len);
```

Comment:
Hmm, can we make this be something like

```suggestion
  return strndup (buf, MIN(len, sizeof (buf)));
```

instead to be safe?

Reply:
why should we? `fread` couldn't return more than `sizeof(buf)`